src/object/srv_obj.c

```diff
@@ -1534,14 +1534,15 @@ ds_obj_rw_handler(crt_rpc_t *rpc)
 
 	/* Stop the distribute transaction */
 	rc = dtx_leader_end(&dlh, ioc.ioc_coc, rc);
-	if (rc == -DER_AGAIN) {
-		if (dlh.dlh_handle.dth_renew) {
-			/* epoch conflict, renew it and retry. */
-			orw->orw_epoch = crt_hlc_get();
-			flags &= ~ORF_RESEND;
-			memset(&dlh, 0, sizeof(dlh));
-			D_GOTO(renew, rc);
-		}
+	if (rc == -DER_TX_RESTART) {
+		D_ASSERT(dlh.dlh_handle.dth_renew);
```

Comment:
maybe we can get rid of this renew flag if these assertions hold.    But I didn't want to make too many changes to this code.

Reply:
Yes, I have a patch (https://github.com/daos-stack/daos/pull/2432) to remove useless DTX flags. I will rebase such patch after this one landed.